include/boost/sml.hpp

```diff
@@ -1703,6 +1703,23 @@ class sm {
     });
     return result;
   }
+  template <class T = aux::identity<sm_t>, class TState>
```

Comment:
It seems like a nice utility :+1: I don't think is required to be part of the sml core, it can be just a free function in the examples or extensions instead as it's implemented using `visit_current_states`